From: Giovanni Mascellani <gio@debian.org>
Subject: Fix other __linux appearances

Fix some remaining appearances of __linux with __linux__, which is
more portable.

diff --git a/src/icu/time_zone.cpp b/src/icu/time_zone.cpp
index 8276231..9032c8d 100644
--- a/libs/locale/src/icu/time_zone.cpp
+++ b/libs/locale/src/icu/time_zone.cpp
@@ -19,7 +19,7 @@
 //
 
 #if U_ICU_VERSION_MAJOR_NUM == 4 && (U_ICU_VERSION_MINOR_NUM * 100 + U_ICU_VERSION_PATCHLEVEL_NUM) <= 402
-# if defined(__linux) || defined(__FreeBSD__) || defined(__APPLE__)
+# if defined(__linux__) || defined(__FreeBSD__) || defined(__APPLE__)
 #   define BOOST_LOCALE_WORKAROUND_ICU_BUG
 # endif
 #endif
diff --git a/src/posix/numeric.cpp b/src/posix/numeric.cpp
index 2bef81e..ffd8152 100644
--- a/libs/locale/src/posix/numeric.cpp
+++ b/libs/locale/src/posix/numeric.cpp
@@ -29,7 +29,7 @@
 #include "all_generator.hpp"
 
 
-#if defined(__linux) || defined(__APPLE__)
+#if defined(__linux__) || defined(__APPLE__)
 #define BOOST_LOCALE_HAVE_WCSFTIME_L
 #endif
 
diff --git a/src/util/numeric.hpp b/src/util/numeric.hpp
index 892427d..d22fd69 100644
--- a/libs/locale/src/util/numeric.hpp
+++ b/libs/locale/src/util/numeric.hpp
@@ -190,7 +190,7 @@ private:
     {
         std::string tz = ios_info::get(ios).time_zone();
         std::tm tm;
-        #if defined(__linux) || defined(__FreeBSD__) || defined(__APPLE__) 
+        #if defined(__linux__) || defined(__FreeBSD__) || defined(__APPLE__) 
         std::vector<char> tmp_buf(tz.c_str(),tz.c_str()+tz.size()+1);
         #endif
         if(tz.empty()) {
@@ -211,7 +211,7 @@ private:
             gmtime_r(&time,&tm);
             #endif
             
-            #if defined(__linux) || defined(__FreeBSD__) || defined(__APPLE__) 
+            #if defined(__linux__) || defined(__FreeBSD__) || defined(__APPLE__) 
             // These have extra fields to specify timezone
             if(gmtoff!=0) {
                 // bsd and apple want tm_zone be non-const
diff --git a/include/boost/test/impl/debug.ipp b/include/boost/test/impl/debug.ipp
index a5e5f6da..fe34bd16 100644
--- a/libs/test/include/boost/test/impl/debug.ipp
+++ b/libs/test/include/boost/test/impl/debug.ipp
@@ -83,7 +83,7 @@ namespace std { using ::memset; using ::sprintf; }
 
 #    include <procfs.h>
 
-#  elif defined(linux) || defined(__linux)
+#  elif defined(linux) || defined(__linux__)
 
 #    define BOOST_LINUX_BASED_DEBUG
 
